llama.h

```diff
@@ -698,6 +704,23 @@ extern "C" {
                                   char * buf,
                                int32_t   length);
 
+    /// Apply chat template and maybe tokenize it. Inspired by hf apply_chat_template() on python.
+    /// Both "model" and "custom_template" are optional, but at least one is required. "custom_template" has higher precedence than "model"
+    /// NOTE: This function only support some known jinja templates. It is not a jinja parser.
+    /// @param custom_template A Jinja template to use for this conversion. If this is nullptr, the model’s default chat template will be used instead.
+    /// @param msg Pointer to a list of multiple llama_chat_message
+    /// @param add_ass Whether to end the prompt with the token(s) that indicate the start of an assistant message.
+    /// @param buf A buffer to hold the output formatted prompt. The recommended alloc size is 2 * (total number of characters of all messages)
+    /// @return The total number of bytes of the formatted prompt. If is it larger than the size of buffer, you may need to re-alloc it and then re-apply the template.
+    LLAMA_API int32_t llama_chat_apply_template(
+              const struct llama_model * model,
+                            const char * custom_template,
```

Comment:
Prefer shorter name `template` instead of `custom_template`:

```suggestion
                            const char * template,
```

Reply:
The compiler isn't happy because `template` is a cpp keyword:

```
In file included from llama.cpp:2:
llama.h:717:42: error: expected ‘,’ or ‘...’ before ‘template’
  717 |                             const char * template,
      |                                          ^~~~~~~~
```

Reply:
Maybe `tmpl` then?

Reply:
`tmpl` is good too. The change is here: https://github.com/ggerganov/llama.cpp/pull/5538/commits/649f6f829057849a3826da89a3a8543f275a0fd0